Gulpfile.js

```diff
@@ -265,6 +278,124 @@ gulp.task('test-client-travis', ['build'], function () {
 
 gulp.task('travis', [process.env.GULP_TASK || '']);
 
+//Documentation
+gulp.task('lint-docs', function () {
+    function lintFiles (files) {
+        return new Promise(function (resolve, reject) {
+            var config = {
+                'files':  files,
+                'config': require('./.markdownlint.json')
+            };
+
+            markdownlint(config, function (err, result) {
+                var lintErr = err || result && result.toString();
+
+                if (lintErr)
+                    reject(lintErr);
+                else
+                    resolve();
+            });
+        });
+    }
+
+    return globby('./docs/articles/**/*.md').then(lintFiles);
+});
+
+gulp.task('clear-site', ['lint-docs'], function (cb) {
+    del('./site/**', function (e) {
+        cb(e);
+    });
+});
+
+gulp.task('fetch-assets-repo', ['clear-site'], function (cb) {
+    git.clone('https://github.com/VasilyStrelyaev/testcafe-gh-page-assets.git', { args: './site' }, function (e) {
+        if (e) return cb(e);
+        cb();
+    });
+});
+
+gulp.task('put-in-articles', ['fetch-assets-repo'], function () {
+    return gulp.src('./docs/articles/**/*')
+		.pipe(gulp.dest('./site/src'));
+});
+
+gulp.task('put-in-navigation', ['put-in-articles'], function () {
```

Comment:
That's what I tried to do at first, but got some strange mess and resorted to sequential execution.
Should have investigated it, sure...